pkg/cloud/vsphere/provisioner/govmomi/create.go

```diff
@@ -495,10 +496,19 @@ func (pv *Provisioner) getCloudProviderConfig(cluster *clusterv1.Cluster, machin
 		return "", err
 	}
 
+	// strip scheme and path, if it is parseable as a URL
+	var server string
+	serverURL, err := url.Parse(clusterConfig.VsphereServer)
+	if err == nil {
+		server = serverURL.Host
+	} else {
+		server = clusterConfig.VsphereServer
```

Comment:
OK, so this is a handler around the way I had things configured in a demo setup, where (a while ago) it seemed that I needed to specify the server as `https://10.117.40.142/sdk` and now for k8s v1.13 `10.117.40.142` is required.  This might be a change in the vsphere-cloud-provider.

url.Parse returns an error if the scheme is missing, i.e. if server=10.117.40.142.  So this assumes that if Parse returns an error the value specified is correct (!).  I should add a comment at least.  

Or not do this transformation at all.  I.e. do some validation earlier.

Reply:
@ntfrnzn i found the root cause of CI failure, url parse does not return error when URL is plain url (no https), and re.Host will be empty, please refer my playground: https://play.golang.org/p/vjnApf0Z4fF

i honestly feel this is a issue at go net/url, but for practical purpose of this PR, you could workaround it by an additional check of `serverURL` and `serverURL.Host`